qiime2/core/archive/provenance_lib/replay.py

```diff
@@ -42,8 +42,6 @@ class ReplayConfig():
         disk in .tsv format.
     use_recorded_metadata : bool
         If True, replay should use the metadata recorded in provenance.
-    pm : PluginManager
-        The active instance of the QIIME 2 PluginManager.
```

Comment:
I think we should probably keep this parameter, it can just be set to `None` by default, in which case it will instantiate. Otherwise, we would probably want to use the "find singleton" classmethod (I forget it's name), that way it will use the pre-instantiated manager. But I think the parameter is a lot clearer.

Reply:
I think the instantiation will return the preexisting instance if it exists:

```python
>>> pm = PluginManager()
>>> pm2 = PluginManager()
>>> id(pm) == id(pm2)
>>> True
```

Reply:
I was thinking of  `qiime2.sdk.PluginManager.reuse_existing()` which will error if there isn't one. But I don't think we actually want that, so the constructor (like in your example) is all we need.